### PR TITLE
feat: cliplugin enable and publish functionality

### PR DESCRIPTION
**Pull request by garyleefight**
Implemented amplify add hosting && amplify hosting publish for both CICD and manual categories.

### CODE REVIEW

File: packages/amplify-console-hosting/utils/path-manager.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
+const path = require('path');
```
Comment (kaustavghosh06) [inline packages/amplify-console-hosting/utils/path-manager.js]: Why do you need to re-mplement this file and not use `context.amplify` instead?

File: packages/amplify-console-hosting/package.json [hunk 2]
```diff
@@ -0,0 +1,25 @@
+{
```
Comment (kaustavghosh06) [inline packages/amplify-console-hosting/package.json]: Can you please take a look at the package json's of other packages in this repo and replicate the scripts?

File: packages/amplify-console-hosting/event-handlers/handle-PostInit.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+const eventName = 'PostInit';
```
Comment (kaustavghosh06) [inline packages/amplify-console-hosting/event-handlers/handle-PostInit.js]: Do you need this file?

File: packages/amplify-console-hosting/event-handlers/handle-PostPush.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+const eventName = 'PostPush';
```
Comment (kaustavghosh06) [inline packages/amplify-console-hosting/event-handlers/handle-PostPush.js]: Do you need this file?

**Comment by kaustavghosh06** [comment]
Please also add E2E tests for this feature.

File: packages/amplify-console-hosting/utils/path-manager.js (hunk 1 above)
Comment (garyleefight) [reply packages/amplify-console-hosting/utils/path-manager.js]: Caused I wants template path and other paths that are not in context.amplify to be reused across packages. Just in consistence, wrap all native path-manager API I used in this utils code as well

File: packages/amplify-console-hosting/package.json (hunk 2 above)
Comment (garyleefight) [reply packages/amplify-console-hosting/package.json]: Will change

File: packages/amplify-console-hosting/event-handlers/handle-PostInit.js (hunk 3 above)
Comment (garyleefight) [reply packages/amplify-console-hosting/event-handlers/handle-PostInit.js]: No I don't use any of them. Can be deleted

File: packages/amplify-console-hosting/event-handlers/handle-PostPush.js (hunk 4 above)
Comment (garyleefight) [reply packages/amplify-console-hosting/event-handlers/handle-PostPush.js]: Same as above

**Pull request closed without merging**